hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java

```diff
@@ -545,6 +552,39 @@ null, raftGroup, createRetryPolicy(conf), tlsConfig, conf)) {
             targetPeerId, reply);
         throw new IOException(reply.getException());
       }
+    } finally {
+      // Raft peers priorities need to be reset regardless of the result
+      // of transfer leadership
+      resetPriorities(conf, raftGroup, tlsConfig);
+    }
+  }
+
+  private static void resetPriorities(ConfigurationSource conf,
+      RaftGroup raftGroup, GrpcTlsConfig tlsConfig) {
+    List<RaftPeer> resetPeers = new ArrayList<>();
+    for (RaftPeer peer : raftGroup.getPeers()) {
+      resetPeers.add(
+          RaftPeer.newBuilder(peer)
+              .setPriority(NEUTRAL_PRIORITY)
+              .build()
+      );
+    }
+    LOG.info("Resetting Raft peers priorities after transfer leadership");
+    try (RaftClient raftClient = newRaftClient(SupportedRpcType.GRPC, null,
+        null, raftGroup, createRetryPolicy(conf), tlsConfig, conf)) {
+      RaftClientReply reply = raftClient.admin().setConfiguration(resetPeers);
+      if (reply.isSuccess()) {
+        LOG.info("Successfully reset priorities for division: {}",
+            resetPeers);
+      } else {
+        LOG.warn("Failed to reset priorities for division: {}." +
+            " Ratis reply: {}", resetPeers, reply);
+      }
+    } catch (IOException e) {
+      LOG.error("Exception thrown when trying to reset priorities for " +
```

Comment:
Is it necessary to prompt User that `resetPriorities` failed by command line output?

Reply:
Thank you for the review. I was thinking about this as well, but I think it's enough to reset priorities to be best effort and just log the error.